tests/func/test_checkout.py

```diff
@@ -409,44 +407,45 @@ def setUp(self):
         self._prepare_repo()
 
     def test(self):
+        encoding = sys.stderr.encoding
         with self._caplog.at_level(logging.INFO, logger="dvc"), patch.object(
-            sys, "stdout"
-        ) as stdout_mock:
-            self.stdout_mock = logger.handlers[0].stream = stdout_mock
+            sys, "stderr"
+        ) as stderr_mock:
+            self.stderr_mock = logger.handlers[0].stream = stderr_mock
+
+            if sys.version_info[:1] < (3,):
+                sys.stderr.encoding = encoding
 
             ret = main(["checkout"])
             self.assertEqual(0, ret)
 
-        stdout_calls = self.stdout_mock.method_calls
+        stdout_calls = self.stderr_mock.method_calls
         write_calls = self.filter_out_non_write_calls(stdout_calls)
         write_calls = self.filter_out_empty_write_calls(write_calls)
         self.write_args = [w_c[1][0] for w_c in write_calls]
 
-        pattern = re.compile(".*\\[.{30}\\].*%.*")
-        progress_bars = [
-            arg
-            for arg in self.write_args
-            if pattern.match(arg) and "unpacked" not in arg
-        ]
-
+        pattern = re.compile(r"%\|\W+\| .*\[.*\]")
+        progress_bars = filter(pattern.search, self.write_args)
+        progress_bars = [arg for arg in progress_bars if "unpacked" not in arg]
         update_bars = progress_bars[:-1]
         finish_bar = progress_bars[-1]
 
-        self.assertEqual(4, len(update_bars))
-        assert re.search(".*\\[#{7} {23}\\] 25%.*", progress_bars[0])
-        assert re.search(".*\\[#{15} {15}\\] 50%.*", progress_bars[1])
-        assert re.search(".*\\[#{22} {8}\\] 75%.*", progress_bars[2])
-        assert re.search(".*\\[#{30}\\] 100%.*", progress_bars[3])
+        # at least the inital (blank) update_bar should be printed;
+        # but maybe no intermediate ones
+        self.assertLessEqual(1, len(update_bars))
+        self.assertLessEqual(len(update_bars), 4)
 
         self.assertCaretReturnFollowsEach(update_bars)
         self.assertNewLineFollows(finish_bar)
 
-        self.assertAnyEndsWith(update_bars, self.FOO)
-        self.assertAnyEndsWith(update_bars, self.BAR)
-        self.assertAnyEndsWith(update_bars, self.DATA)
-        self.assertAnyEndsWith(update_bars, self.DATA_SUB)
-
-        self.assertTrue(finish_bar.endswith("Checkout finished!"))
+        # self.assertAnyEndsWith(update_bars, self.FOO)
```

Comment:
Please don't leave commented-out code :slightly_smiling_face:

Reply:
It does seem like we could throw out this test, right? Please feel free to remove tests that you don't see the need in because they are already tested by tqdm.